plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -114,7 +114,7 @@ func Run(s *options.SchedulerServer) error {
 	rl := &resourcelock.EndpointsLock{
 		EndpointsMeta: metav1.ObjectMeta{
 			Namespace: "kube-system",
-			Name:      "kube-scheduler",
+			Name:      s.SchedulerName,
```

Comment:
default the lock object to `kube-scheduler` in the `kube-system` namespace, let them pick a different namespace and/or name for the lock object